### PR TITLE
homebrew generic mac port

### PR DESCRIPTION
**Pull request by jedwards4b**
Provides a new machine definition 'homebrew' that should work on any mac system without
modification of cime source.   The user must install homebrew gcc, netcdf and mpich.
The user is also required to create a directory $HOME/projects/scratch
I also realphabetized config_machines.xml

Test suite: cime_developer
Test baseline: 
Test namelist changes: 
Test status: openmp builds fail 

Fixes [CIME Github issue #]

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:

### CODE REVIEW

**Comment by billsacks** [comment]
Before I test this out, can you clarify the intended usage, so that I can test it according to the intended usage?

(1) Is the intended usage that the user will copy this into .cime and customize it, or that they'll use it unmodified with the `--machine homebrew` argument to create_newcase? Or are both of these equally valid ways to use this?

(2) Is this expected to work without any config_compilers.xml file in .cime? If so, is it expected that NETCDF_PATH (or something similar) is defined in your environment?

**Comment by jedwards4b** [comment]
Both are valid.
If netcdf was installed from homebrew you should not need any changes.

File: config/cesm/machines/config_machines.xml [hunk 1]
```diff
@@ -907,70 +882,42 @@
...
+    <CCSM_CPRNC>$CIMEROOT/tools/cprnc/build/cprnc</CCSM_CPRNC>
+    <GMAKE>make</GMAKE>
+    <GMAKE_J>4</GMAKE_J>
+    <BATCH_SYSTEM>none</BATCH_SYSTEM>
+    <SUPPORTED_BY>__YOUR_NAME_HERE__</SUPPORTED_BY>
+    <MAX_TASKS_PER_NODE>4</MAX_TASKS_PER_NODE>
+    <PES_PER_NODE>4</PES_PER_NODE>
     <mpirun mpilib="default">
-      <executable>aprun</executable>
+      <executable>mpirun</executable>
```
Comment (jedwards4b) [inline config/cesm/machines/config_machines.xml]: nope

**Comment by billsacks** [comment]
Okay, looks good - thanks. I'll just await a review from @mvertens if she wants to... if not, I'll merge it.

**Comment by jedwards4b** [comment]
@mvertens still waiting for your review.

**Comment by mvertens** [comment]
I'm fine with the review from @billsacks as it is.

**Pull request merged**